tests/framework/cli/test_cli.py

```diff
@@ -104,9 +104,13 @@ def test_print_version(self):
         assert result_abr.exit_code == 0
         assert version in result_abr.output
 
+    def test_info(self):
+        result = CliRunner().invoke(cli, ["info"])
+        assert result.exit_code == 0
+
```

Comment:
This will fail properly even if the mocking is wrong.

Reply:
I don't think this is doing anything. We already have `test_info_no_plugins` (which actually shouldn't need the `entry_points` fixture it's using), and that does the same as your test. Unfortunately it can't pick up on the bug because the error is only triggered if there are plugins installed, and when this test runs there aren't any so it passes already.

Reply:
@AntonyMilneQB Thanks, I missed the test. Actually, the test would fail properly without any change now.

The reason is exactly the problem we discussed. The `MagicMock` should throw `AttributeError` when accessing attributes that do not exist.  `entry_point.module_name` would result in `AttributeError` now due to the change in `conftest.py`.

Reply:
I think best thing is to make a new `info.feature` e2e test. If you search for `I have installed the test plugin` you'll see we already have a step which will help create a project that does has a plugin installed. If you run `kedro info` on such a project that e2e test would pick up on the bug properly.

Reply:
@noklam hmmm, how can `test_info_no_plugins` pick up the problem when there's no plugins in that test so it doesn't go into the for loop that triggers the exception?

Reply:
See the test result below. It can because it doesn't really load any plugin. We mock the `entry_point` so it always return a list of `[ep]` that goes into the loop. If you look at the test `test_info_with_plugin`, it does the same thing, it just also mock the `entry_point.module` to fake the attribute.

If you are saying the loop was never entered then both test would not be valid.


_______________________________________________________________________________________________________________________________________________ TestCliCommands.test_info_no_plugins ________________________________________________________________________________________________________________________________________________

self = <tests.framework.cli.test_cli.TestCliCommands object at 0x7fdab0b8ecd0>

    @mark.usefixtures("entry_point")
    def test_info_no_plugins(self):
        result = CliRunner().invoke(cli, ["info"])
>       assert result.exit_code == 0
E       assert 1 == 0
E        +  where 1 = <Result AttributeError("Mock object has no attribute 'module_name'")>.exit_code

Reply:
We can have a test that mocks for no plugins and one without mocking, but that feels a bit duplicate to me.

Could we just remove the mocking for `test_info_no_plugins` so it basically acts the same as e2etest?

I think it's still valid to add the e2etest to make sure the environment is correct, but a unit test will still be valuable because it's fast to run and easy to debug.